### PR TITLE
Celebrate Dia de Muertos 🌺💀🌹 

### PR DESCRIPTION
**Pull request by petrosagg**
### Motivation

Vanilla rustc does not do dead-code analysis cross-crate. I used an external program[1] to analyze our codebase for dead code. The library produces a ton of false positives so this PR includes things I manually deemed worth to remove.

[1] https://github.com/est31/warnalyzer

### CODE REVIEW

File: src/expr/src/relation/mod.rs [hunk 1]
```diff
@@ -1298,14 +1297,6 @@ impl MirRelationExpr {
         .unwrap()
     }
 
-    /// Pretty-print this MirRelationExpr to a string.
```
Comment (asenac) [inline src/expr/src/relation/mod.rs:1301]: These `pretty_*` ones are handy when doing `println`-based debugging of the transforms.

Comment (petrosagg) [reply src/expr/src/relation/mod.rs]: thanks for getting through them! re-added

**Comment by frankmcsherry** [comment]
Most of the things that I wrote that are in here were written for a future reason. It would be great not to have to retype them. Maybe the removals should be opt in for the crate owners?

**Comment by petrosagg** [comment]
@frankmcsherry yes this PR is meant to be vetted by the respective crate owners. If you'd rather have a dataflow-specific PR with the functions that were flagged as dead code that's easy to do

**Comment by petrosagg** [comment]
superseded by:

https://github.com/MaterializeInc/materialize/pull/8968
https://github.com/MaterializeInc/materialize/pull/8967
https://github.com/MaterializeInc/materialize/pull/8966
https://github.com/MaterializeInc/materialize/pull/8965
https://github.com/MaterializeInc/materialize/pull/8964

**Comment by ruchirK** [comment]
I love the emojis on this pr.

**Pull request closed without merging**